Tightening Up Intro (2) (#26)
lamps-wg/kyber-certificates: draft-ietf-lamps-kyber-certificates.md

    email: bas@westerbaan.name

normative:
  X680:
    target: https://www.itu.int/rec/T-REC-X.680
    title: "Information technology - Abstract Syntax Notation One (ASN.1): Specification of basic notation"
    date: Feburary 2021
    author:
      org: ITU-T
    seriesinfo:
        ISO/IEC: 8824-1:2021
  X690:
    target: https://www.itu.int/rec/T-REC-X.690
    title: "Information technology - Abstract Syntax Notation One (ASN.1): ASN.1 encoding rules: Specification of Basic Encoding Rules (BER), Canonical Encoding Rules (CER) and Distinguished Encoding Rules (DER)"
    date: Feburary 2021
    author:
      org: ITU-T
    seriesinfo:
        ISO/IEC: 8825-1:2021

> informative:

Author: ```suggestion
informative:
  PQCProj:
    target: https://csrc.nist.gov/projects/post-quantum-cryptography
    title: Post-Quantum Cryptography Project
    author:
      - org: National Institute of Standards and Technology
    date: 2016-12-20
```

# Introduction

> Kyber is a key-encapasulation mechanism (KEM). This document specifies

Reviewer: If you keep this intro, just mention the word NIST PQ standardization effort. The rest is fine. Something like 
> [...] key-encapasulation mechanism (KEM) standardized in the US NIST PQC Project {{PQCProj}}

> Kyber is a key-encapasulation mechanism (KEM). This document specifies
> the use of the Kyber algorithm at three security levels, Kyber512,
> Kyber768, and Kyber1024, in X.509 public key certifiates; see
> {{!RFC5280}}. Public and private key encodings are alsospecified.

Author: ```suggestion
Kyber is a key-encapsulation mechanism (KEM) standardized by the US NIST
PQC Project {{PQCProj}}. This document specifies the use of the Kyber
algorithm at three security levels, Kyber512, Kyber768, and Kyber1024,
in X.509 public key certificates; see {{!RFC5280}}. Public and private
key encodings are also specified.
```

## ASN.1 and Kyber Identifiers

An ASN.1 module {{X680}} is included for reference purposes. Note that
as per {{RFC5280}}, certificates use the Distinguished Encoding Rules;
see {{X690}}. Also note that NIST defined the object identifiers for
the Kyber algorithms in an ASN.1 modulle; see (TODO insert reference).

## Applicability Statement

Kyber certificates are used in protocols where the public key is used to

> encapsulate a shared secret used to derive a symmetric key used to

Reviewer: Same as the other comment. Kyber doesn't allow you to pick the shared secret to encapsulate. Instead, perhaps: "[...] to generate and encapsulate [...]"

Reviewer 2: nit: add a colon to start of list?
>This document specifies the use of the Kyber algorithm at three security levels: Kyber512, Kyber768, and Kyber1024,

> encapsulate a shared secret used to derive a symmetric key used to

Author: ```suggestion
generate and encapsulate a shared secret used to derive a symmetric key used to
```

# Introduction

Kyber is a key-encapsulation mechanism (KEM) standardized by the US NIST
PQC Project {{PQCProj}}. This document specifies the use of the Kyber

> algorithm at three security levels, Kyber512, Kyber768, and Kyber1024,

Author: ```suggestion
algorithm at three security levels: Kyber512, Kyber768, and Kyber1024,
```

# Key Usage Bits

The intended application for the key is indicated in the keyUsage

> certificate extension; see {{See Section 4.2.1.3 of RFC5280}}.

Author: ```suggestion
certificate extension; see {{Section 4.2.1.3 of RFC5280}}.
```
